utils/contract-wrapper.ts

```diff
@@ -1,47 +1,43 @@
-import {PriceFeed} from "../typechain/PriceFeed";
-import {ContractInterface, ethers, Signer} from "ethers";
-import {PriceFeed__factory} from "../typechain/factories/PriceFeed__factory";
+import { ethers, Signer } from "ethers";
+import { PriceFeed__factory } from "../typechain/factories/PriceFeed__factory";
+
+// TODO: Refactor this file and add many comments to make it more clear
 
 const { getSignedPrice } = require("../utils/api-connector");
 
 export type PriceDataType = {
     symbols: string[],
     values: number[],
     timestamp: number
-}
+};
 
 export type SignedPriceDataType = {
     priceData: PriceDataType,
     signer: string,
     signature: string
-}
+};
 
-
-async function getPriceData(signer:Signer, dataProvider:string, asset?: string) {
-    let {priceData, signature} = await getSignedPrice(dataProvider, asset);
+async function getPriceData(signer: Signer, dataProvider: string, asset?: string) {
+    let { priceData, signature } = await getSignedPrice(dataProvider, asset);
 
     let priceFeed = PriceFeed__factory.connect(ethers.constants.AddressZero, signer);
     let setPriceTx = await priceFeed.populateTransaction.setPrices(priceData, signature);
     
-    let setPriceData = setPriceTx.data ? setPriceTx.data.substr(2) : "";
+    const setPriceData = remove0xFromHexString(setPriceTx.data, {
+      allowFalsyValues: true,
+    });
 
-    let clearPriceTx = await priceFeed.populateTransaction.clearPrices(priceData);
-    let clearPricePrefix = clearPriceTx.data ? clearPriceTx.data.substr(2,8) : "";
+    let clearPriceTx = await priceFeed.populateTransaction.clearPrices(priceData); // priceData may have any value, we don't use it
+    let clearPricePrefix = clearPriceTx.data ? clearPriceTx.data.substr(2,8) : ""; // we get 4 bytes from 2nd to 5th (skipping 1st (0x)) - this is the function signature clearPrices
 
-    //Add priceDataLen info
-    let priceDataLen = setPriceData.length/2;
-    console.log("Price data len: " + priceDataLen);
+    // Add priceDataLen info
+    const priceDataLen = countBytesInHexString(setPriceData);
 
-    return clearPricePrefix + setPriceData + priceDataLen.toString(16).padStart(4, "0");
+    // Template of data that we add to the end of tx is below
+    // [CLEAR_PRICE_FUNCTION_SIGNATURE (4 bytes)] + [SET_PRICE_FUNCTION_WITH_ARGUMENTS (Variable bytes size, 4 bytes for signature + variable length bytes for arguments. It is callable)] + [SET_PRICE_FUNCTION_WITH_ARGS_BYTE_LENGTH (2 bytes)]
```

Comment:
How about adding this as a comment to show structurre:
[SIG_CLEAR|4][SIG_SET|4][DATA|...][DATA_LEN|2][MARKER|32]
and then explaining each part below